src/AutoRest.CSharp/LowLevel/Output/ClientFields.cs

```diff
@@ -31,7 +31,7 @@ internal class ClientFields : IReadOnlyCollection<FieldDeclaration>
 
         public ClientFields(BuildContext<LowLevelOutputLibrary> context, IEnumerable<Parameter> parameters)
         {
-            ClientDiagnosticsField = new(Private | ReadOnly, typeof(ClientDiagnostics), "_" + KnownParameters.ClientDiagnostics.Name);
+            ClientDiagnosticsField = new(Internal | ReadOnly, typeof(ClientDiagnostics), "_" + KnownParameters.ClientDiagnostics.Name);
```

Comment:
We also need to convert it to a property, so:
```cs
ClientDiagnosticsProperty = new(Internal | ReadOnly, typeof(ClientDiagnostics), KnownParameters.ClientDiagnostics.Name.FirstCharToUpperCase(), writeAsProperty: true);
```

Reply:
Hi Alexander, Do you want to generate code as following:
![image](https://user-images.githubusercontent.com/10458143/150506283-c95ea63f-fc01-4b92-9a86-3e8d64012de8.png)

Reply:
Yep, looks good to me. @annelo-msft, do you know if internal property has to be virtual?

Reply:
Just one more thing. We usually have properties after fields, so let's add it to the `fields` list after all parameters are added:
https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/LowLevel/Output/ClientFields.cs#L72

Reply:
> @annelo-msft, do you know if internal property has to be virtual?

The reason we make public members virtual is to enable mocking.  Since an internal field isn't visible outside an assembly, I don't think it's possible to mock it, so no, I don't think it needs to be virtual.

Reply:
@annelo-msft @AlexanderSher Can the property name is the same as its type?

Reply:
Yes, it can.  Are you seeing any problems with this?

Reply:
No. I don't see any functionality problem. I just ask to confirm if it will break any naming convention. Great! If is ok with this, Is there any problem with this PR? If no, would you please approve it. Thanks